store/store.go

```diff
@@ -43,8 +43,8 @@ const (
 	blobType int64 = iota
 	imageManifestType
 
-	defaultPathPerm os.FileMode = 0777
-	defaultFilePerm os.FileMode = 0660
+	defaultPathPerm = os.FileMode(0770 | os.ModeSetgid)
+	defaultFilePerm = os.FileMode(0660 | os.ModeSetgid)
```

Comment:
It makes even less sense for files considering that they're not even executable by permission (0660), so I think we can safely remove the OR-addition for defaultFilePerm.